### PR TITLE
Sized tiles

### PR DESCRIPTION
**Pull request by Quarris**
This PR implements Chiselling of tiles. The tiles can be chiselled at each corner to create various shapes.
Water respects the chiselled tiles and can pass through if the appropriate corner has been chiselled.
This also fixes the liquid containers consuming water in a weird way to make it lose its contents.
Tiles can now determines from which side the water can spread in and out of. For example the door can spread only from the opposite side of how it is placed, so if the door is on the left, the water can spread from the right.

TODO:
- There is a bug with liquid disappearing when it is inside of a chiselled tile and the tile is chiselled again. Unknown fix right now, at least it is not magically generating water.
- Now that sized tiles are a thing, I will start working on the stepping up so that movement is smooth across the world.

[API PR](https://github.com/RockBottomGame/API/pull/42)

### CODE REVIEW

File: src/main/java/de/ellpeck/rockbottom/world/tile/TileWoodDoor.java [hunk 1]
```diff
@@ -96,6 +99,13 @@ public boolean isFullTile() {
         return false;
     }
 
+    @Override
+    public boolean canLiquidSpread(IWorld world, int x, int y, TileLiquid liquid, Direction dir) {
+        TileState state = world.getState(x, y);
+        boolean facingRight = state.get(StaticTileProps.FACING_RIGHT);
+        return state.get(StaticTileProps.OPEN) || facingRight && dir == Direction.LEFT || !facingRight && dir == Direction.RIGHT;
```
Comment (raphydaphy) [inline src/main/java/de/ellpeck/rockbottom/world/tile/TileWoodDoor.java]: i think we are doing `return state.get(StaticTileProps.OPEN) || (facingRight && dir == Direction.LEFT) || (!facingRight && dir == Direction.RIGHT);`

Comment (Quarris) [reply src/main/java/de/ellpeck/rockbottom/world/tile/TileWoodDoor.java]: Yes that is what we are doing

File: src/main/java/de/ellpeck/rockbottom/item/ItemBowl.java [hunk 2]
```diff
@@ -17,6 +17,6 @@ public boolean allowsLiquid(ItemInstance instance, ResourceName name) {
 
     @Override
     public int getCapacity(ItemInstance instance) {
-        return 5;
+        return 6;
```
Comment (canitzp) [inline src/main/java/de/ellpeck/rockbottom/item/ItemBowl.java]: Why this change?

Comment (Quarris) [reply src/main/java/de/ellpeck/rockbottom/item/ItemBowl.java]: So that bowl take half a tiles worth of liquid as initially intended

File: src/main/java/de/ellpeck/rockbottom/world/tile/TileWoodDoor.java (hunk 1 above)
Comment (raphydaphy) [reply src/main/java/de/ellpeck/rockbottom/world/tile/TileWoodDoor.java]: But that's not what you did

File: src/main/java/de/ellpeck/rockbottom/item/ItemBowl.java (hunk 2 above)
Comment (canitzp) [reply src/main/java/de/ellpeck/rockbottom/item/ItemBowl.java]: ok

File: src/main/java/de/ellpeck/rockbottom/world/tile/TileWoodDoor.java (hunk 1 above)
Comment (Quarris) [reply src/main/java/de/ellpeck/rockbottom/world/tile/TileWoodDoor.java]: what? Yes it is.

File: src/main/java/de/ellpeck/rockbottom/world/tile/TileWoodDoor.java [hunk 3]
```diff
@@ -7,13 +7,16 @@
...
+import de.ellpeck.rockbottom.api.tile.TileLiquid;
+import de.ellpeck.rockbottom.api.tile.state.BoolProp;
 import de.ellpeck.rockbottom.api.tile.state.TileState;
 import de.ellpeck.rockbottom.api.util.BoundBox;
 import de.ellpeck.rockbottom.api.util.Direction;
 import de.ellpeck.rockbottom.api.util.reg.ResourceName;
 import de.ellpeck.rockbottom.api.world.IWorld;
 import de.ellpeck.rockbottom.api.world.layer.TileLayer;
 import de.ellpeck.rockbottom.render.tile.TileDoorRenderer;
+import org.omg.CORBA.TIMEOUT;
```
Comment (raphydaphy) [inline src/main/java/de/ellpeck/rockbottom/world/tile/TileWoodDoor.java]: what is this import 😨
